swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -139,85 +149,95 @@ class Adapter {
     }
   }
 
-  /// Currently disabled resources
+  /// Internet resource enabled state
   private var internetResourceEnabled: Bool
 
-  /// Cache of internet resource
-  private var internetResource: Resource?
-
-  /// Keep track of resources
+  /// Keep track of resources for UI
   private var resourceListJSON: String?
 
   /// Starting parameters
   private let apiURL: String
   private let token: Token
-  private let id: String
+  private let deviceId: String
   private let logFilter: String
-  private let connlibLogFolderPath: String
+  private let accountSlug: String
 
-  init(
-    apiURL: String,
-    token: Token,
-    id: String,
-    logFilter: String,
-    accountSlug: String,
-    internetResourceEnabled: Bool,
-    packetTunnelProvider: PacketTunnelProvider
-  ) {
-    self.apiURL = apiURL
-    self.token = token
-    self.id = id
-    self.packetTunnelProvider = packetTunnelProvider
-    self.callbackHandler = CallbackHandler()
-    self.logFilter = logFilter
-    self.accountSlug = accountSlug
-    self.connlibLogFolderPath = SharedAccess.connlibLogFolderURL?.path ?? ""
-    self.networkSettings = nil
-    self.internetResourceEnabled = internetResourceEnabled
-  }
+  func start() throws {
+    Log.log("Adapter.start: Starting session for account: \(accountSlug)")
 
-  // Could happen abruptly if the process is killed.
-  deinit {
-    Log.log("Adapter.deinit")
+    // Get device metadata
+    let deviceName = DeviceMetadata.getDeviceName()
+    let osVersion = DeviceMetadata.getOSVersion()
+    let deviceInfo = try JSONEncoder().encode(DeviceMetadata.deviceInfo())
+    let deviceInfoStr = String(data: deviceInfo, encoding: .utf8) ?? "{}"
+    let logDir = SharedAccess.connlibLogFolderURL?.path ?? "/tmp/firezone"
 
-    // Cancel network monitor
-    networkMonitor?.cancel()
-  }
+    // Create the session
+    let session: Session
+    do {
+      session = try Session.newApple(
+        apiUrl: apiURL,
+        token: token.description,
+        deviceId: deviceId,
+        accountSlug: accountSlug,
+        deviceName: deviceName,
+        osVersion: osVersion,
+        logDir: logDir,
+        logFilter: logFilter,
+        deviceInfo: deviceInfoStr,
+        isInternetResourceActive: internetResourceEnabled
+      )
+    } catch {
+      throw AdapterError.connlibConnectError(String(describing: error))
+    }
 
-  /// Start the tunnel.
-  func start() throws {
-    Log.log("Adapter.start")
+    // Create channels - following Rust pattern with separate sender/receiver
+    let (commandSender, commandReceiver): (Sender<SessionCommand>, Receiver<SessionCommand>) =
+      Channel.create()
+    self.commandSender = commandSender
+
+    let (eventSender, eventReceiver): (Sender<Event>, Receiver<Event>) = Channel.create()
+    self.eventReceiver = eventReceiver
```

Comment:
Do we need to store the event receiver in `self`? I think we could just pass it on as a local variable to the task that processes it. That should give us one less thing to clean up.